pom.xml

```diff
@@ -118,6 +118,11 @@
         </dependency>
         <dependency>
             <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-configuration-processor</artifactId>
```

Comment:
This is not IDE-specific, that was just an example. The meta-data files are present in all spring libraries and this is an official feature. See [here](http://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html#configuration-metadata-annotation-processor).